sdk/python/feast/infra/offline_stores/contrib/trino_offline_store/connectors/upload.py

```diff
@@ -0,0 +1,189 @@
+    pyarrow_schema = pyarrow_schema_from_dataframe(df=df)
+
+    def _is_nan(value: Any) -> bool:
+        if value is None:
+            return True
+
+        try:
+            return np.isnan(value)
+        except TypeError:
+            return False
+
+    def _format_value(row: pd.Series, schema: Dict[str, Any]) -> str:
+        formated_values = []
+        for row_name, row_value in row.iteritems():
+            if schema[row_name].startswith("timestamp"):
+                if isinstance(row_value, datetime):
+                    row_value = format_datetime(row_value)
+                formated_values.append(f"TIMESTAMP '{row_value}'")
+            elif isinstance(row_value, list):
+                formated_values.append(f"ARRAY{row_value}")
+            elif isinstance(row_value, str):
+                formated_values.append(f"'{row_value}'")
+            elif _is_nan(row_value):
+                formated_values.append("NULL")
+            else:
+                formated_values.append(f"{row_value}")
+
+        return f"({','.join(formated_values)})"
+
+    results = df.apply(_format_value, args=(pyarrow_schema,), axis=1).tolist()
+    return ",".join(results)
+
+
+def format_datetime(t: datetime) -> str:
+    if t.tzinfo:
+        t = t.astimezone(tz=utc)
+    return t.strftime("%Y-%m-%d %H:%M:%S.%f")
+
+
+def upload_pandas_dataframe_to_trino(
+    client: Trino,
+    df: pd.DataFrame,
+    table: str,
+    connector_args: Optional[Dict[str, Any]] = None,
+) -> None:
+    connector_args = connector_args or {}
+    connector_name = connector_args["type"]
+
+    if connector_name in CONNECTORS_DONT_SUPPORT_CREATE_TABLE:
+        raise ValueError(
+            f"The connector '{connector_name}' is not supported because it does not support CREATE TABLE operations"
+        )
+    elif connector_name in CONNECTORS_WITHOUT_WITH_STATEMENTS:
+        with_statement = ""
+    elif connector_name in {"hive", "iceberg"}:
+        if "file_format" not in connector_args.keys():
+            raise ValueError(
+                f"The connector {connector_name} needs the argument 'file_format' in order to create tables with Trino"
+            )
+        with_statement = f"WITH (format = '{connector_args['file_format']}')"
+    elif connector_name in {"kudu", "phoenix", "sqlserver"}:
+        raise ValueError(
+            f"The connector {connector_name} is not yet supported. PRs welcome :)"
+        )
+    else:
+        raise ValueError(
+            f"The connector {connector_name} is not part of the connectors listed in the Trino website: https://trino.io/docs/current/connector.html"
+        )
+
+    client.execute_query(
+        CREATE_SCHEMA_QUERY_TEMPLATE.format(
+            table=table,
+            schema=trino_table_schema_from_dataframe(df=df),
+            with_statement=with_statement,
+        )
+    )
+
+    # Upload batchs of 1M rows at a time
+    for batch_df in pandas_dataframe_fix_batches(df=df, batch_size=1000000):
```

Comment:
As per comment above
```suggestion
    for batch_df in pandas_dataframe_fix_batches(df=df, batch_size= batch_size):
```